enumfields/fields.py

```diff
@@ -34,7 +36,11 @@ def to_python(self, value):
         raise ValidationError('%s is not a valid value for enum %s' % (value, self.enum), code="invalid_enum_value")
 
     def get_prep_value(self, value):
-        return None if value is None else self.enum(value).value
+        if value is None:
+            return None
+        if not isinstance(value, self.enum):
+            return self.enum(value).value
+        return value.value  # Already the correct type
```

Comment:
@suutari-ai: I instrumented my venv's django-enumfields to keep track of which code path it used and ran the test suite for shuup/shuup@abcdcd414b1895f3198a7951e267cde33ec9d796 (nice hash!):

```
prep_value benchmark: [('nocast', 114)]
```

I think it's safe to say the no-cast case is somewhat more common 😆